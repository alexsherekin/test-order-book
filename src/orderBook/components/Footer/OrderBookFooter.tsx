import React, { FC } from "react";
import Button from "react-bootstrap/Button";
import styles from "./OrderBookFooter.module.css";

export interface OrderBookFooterProps {
  isConnected: boolean;
  toggleFeedClicked: () => void;
  killReconnectFeedClicked: () => void;
};

export const OrderBookFooter: FC<OrderBookFooterProps> = (props) => {
  return (
    <div className={styles.footer}>
      <Button className={styles.button} variant="primary" onClick={props.toggleFeedClicked}>Toggle feed</Button>
      <Button className={styles.button} variant="danger" onClick={props.killReconnectFeedClicked}>
        {props.isConnected ? "Kill feed" : "Reconnect"}
      </Button>
    </div>
  );
};
