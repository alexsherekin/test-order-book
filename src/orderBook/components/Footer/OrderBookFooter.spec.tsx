import React from "react";
import {render} from '@testing-library/react';
import {OrderBookFooter, OrderBookFooterProps} from "./OrderBookFooter";

const renderComponent = (props: OrderBookFooterProps) => {
  return render(<OrderBookFooter {...props} />);
}

describe(OrderBookFooter.name, () => {
  it.each([
    [
      {
        isConnected: true,
        toggleFeedClicked(){},
        killReconnectFeedClicked(){},
      }
    ],
    [
      {
        isConnected: false,
        toggleFeedClicked(){},
        killReconnectFeedClicked(){},
      },
    ]
  ])("should render", (props) => {
    const renderedComponent = renderComponent(props);

    expect(renderedComponent.asFragment()).toMatchSnapshot();
  });
});
