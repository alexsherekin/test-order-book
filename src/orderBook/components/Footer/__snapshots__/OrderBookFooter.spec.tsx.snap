// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBookFooter should render 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <button
      class="button btn btn-primary"
      type="button"
    >
      Toggle feed
    </button>
    <button
      class="button btn btn-danger"
      type="button"
    >
      Kill feed
    </button>
  </div>
</DocumentFragment>
`;

exports[`OrderBookFooter should render 2`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <button
      class="button btn btn-primary"
      type="button"
    >
      Toggle feed
    </button>
    <button
      class="button btn btn-danger"
      type="button"
    >
      Reconnect
    </button>
  </div>
</DocumentFragment>
`;
